python/ray/tune/tests/test_relative_local_dir.py

```diff
@@ -0,0 +1,137 @@
+# coding: utf-8
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import shutil
+import unittest
+
+import ray
+from ray import tune
+from ray.rllib import _register_all
+
+
+class TuneRelativeLocalDirTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.current_dir = os.path.abspath(".")
+        ray.init(num_cpus=1, num_gpus=0, local_mode=True)
+        _register_all()
+
+    @classmethod
+    def tearDownClass(self):
+        ray.shutdown()
+
+    def tearDown(self):
+        os.chdir(self.current_dir)  # Turn the current_dir back
+
+    def testDottedRelativePath(self):
+        local_dir = "test_dotted_relative_local_dir"
+        local_dir = os.path.join(self.current_dir, local_dir)
+
+        tune.run(
+            "PG",
+            name="TuneDottedRelativeLocalDirTest",
+            stop={"training_iteration": 1},
+            checkpoint_freq=1,
+            local_dir="./test_dotted_relative_local_dir",
+            config={
+                "env": "CartPole-v0",
+            })
+
+        self.assertTrue(os.path.isdir(local_dir))
+        expdir = os.path.join(local_dir, "TuneDottedRelativeLocalDirTest")
+        self.assertTrue(os.path.isdir(expdir))
+        trial_dir = None
+        for i in os.listdir(expdir):
```

Comment:
I think a bit more pythonic way of doing this would be:
```python
trial_dir = next(
    child_dir for child_dir in os.listdir(experiment_dir)
    if (os.path.isdir(os.path.join(experiment_dir, child_dir))
        and child_dir.startswith("PG")))
```

Same thing below.